lib/load.php

```diff
@@ -101,14 +101,12 @@ function gutenberg_is_experiment_enabled( $name ) {
 require dirname( __FILE__ ) . '/compat.php';
 require dirname( __FILE__ ) . '/utils.php';
 
-// Include FSE related files only if the experiment is enabled.
-if ( gutenberg_is_experiment_enabled( 'gutenberg-full-site-editing' ) ) {
-	require dirname( __FILE__ ) . '/templates.php';
-	require dirname( __FILE__ ) . '/template-parts.php';
-	require dirname( __FILE__ ) . '/template-loader.php';
-	require dirname( __FILE__ ) . '/edit-site-page.php';
-	require dirname( __FILE__ ) . '/edit-site-export.php';
-}
+require dirname( __FILE__ ) . '/full-site-editing.php';
```

Comment:
Will loading these all the time create any issues for non-FSE themes? Should we just do a conditional `gutenberg_is_experiment_enabled( 'gutenberg-full-site-editing' ) || gutenberg_is_fse_theme()` for this? 

It seems easier than all of the bail early checks within the loaded files.

Reply:
The `locate_template` function (gutenberg_is_fse_theme) triggers an error when called before init, that's why I removed the check.

Reply:
If I recall, I moved all of these to be conditionally loaded because if the WP_Template CPT exists, then certain template resolution code starts working automatically. This meant that anyone could create WP_Template CPT, and start using template resolution behavior on the front end regardless of whether the experiment was enabled, and regardless of the theme.

Basically, are we sure that the new theme check prevents that scenario?

Reply:
Yes, we're sure I tested but regardless, with the change here https://github.com/WordPress/gutenberg/pull/26500#discussion_r515128631 we can restore this.

Reply:
It turns out that even with the new implementation, it's too early to call the function. I think it doesn't matter anyway since all the hooks do early returns if it's not an FSE theme.